dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/indicator/IndicatorTypeController.java

```diff
@@ -27,17 +27,56 @@
  */
 package org.hisp.dhis.webapi.controller.indicator;
 
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.hisp.dhis.common.OpenApi;
+import org.hisp.dhis.dxf2.webmessage.WebMessage;
+import org.hisp.dhis.dxf2.webmessage.WebMessageUtils;
+import org.hisp.dhis.feedback.ConflictException;
+import org.hisp.dhis.feedback.MergeReport;
 import org.hisp.dhis.indicator.IndicatorType;
+import org.hisp.dhis.merge.MergeParams;
+import org.hisp.dhis.merge.MergeService;
+import org.hisp.dhis.merge.MergeType;
+import org.hisp.dhis.merge.indicator.IndicatorTypeMergeParamsProcessor;
 import org.hisp.dhis.schema.descriptors.IndicatorTypeSchemaDescriptor;
 import org.hisp.dhis.webapi.controller.AbstractCrudController;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
 import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
 
 /**
  * @author Morten Olav Hansen <mortenoh@gmail.com>
  */
 @OpenApi.Tags("metadata")
 @Controller
 @RequestMapping(value = IndicatorTypeSchemaDescriptor.API_ENDPOINT)
-public class IndicatorTypeController extends AbstractCrudController<IndicatorType> {}
+@RequiredArgsConstructor
+@Slf4j
+public class IndicatorTypeController extends AbstractCrudController<IndicatorType> {
+
+  private final MergeService indicatorTypeMergeService;
+  private final IndicatorTypeMergeParamsProcessor mergeParamsProcessor;
+
+  @ResponseStatus(HttpStatus.OK)
+  @PreAuthorize("hasRole('ALL') or hasRole('F_INDICATOR_TYPE_MERGE')")
+  @PostMapping(value = "/merge", produces = APPLICATION_JSON_VALUE)
+  public @ResponseBody WebMessage mergeIndicatorTypes(@RequestBody MergeParams params)
+      throws ConflictException {
+    log.info("Indicator type merge received");
+
+    MergeReport report =
+        mergeParamsProcessor.processMergeParams(
+            indicatorTypeMergeService, params, MergeType.INDICATOR_TYPE);
```

Comment:
The `IndicatorTypeMergeParamsProcessor` is a spring component so the service should not be passed as a parameter here but be injected by spring when constructing the component.

Reply:
rejigged wiring